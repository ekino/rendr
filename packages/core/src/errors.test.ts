import {
  NotFoundError,
  InternalServerError,
  NormalizationError,
  RendrError
} from "./errors";

const errors = [
  {
    class: NotFoundError,
    message: "Not Found Error",
    name: "NotFoundError"
  },
  {
    class: InternalServerError,
    message: "Internal Server Error",
    name: "InternalServerError"
  },
  {
    class: NormalizationError,
    message: "Normalization Error",
    name: "NormalizationError"
  }
];

describe("test errors", () => {
  errors.forEach(error => {
    it(`${error.message} > should return default message`, () => {
      const err = new error.class();

      expect(err instanceof RendrError).toBeTruthy();
      expect(err instanceof error.class).toBeTruthy();
      expect(err instanceof Error).toBeTruthy();
      expect(err.stack).toBeDefined();
      expect(err.toString()).toEqual(`${error.name}: ${error.message}`);
    });

    it(`${error.message} > should return custom message`, () => {
      const err = new error.class("custom message");
      expect(err.message).toEqual(`custom message`);
      expect(err).toMatchSnapshot();
    });

    it(`${error.message} > should return previous error, with default message`, () => {
      const previousError = new Error("My Error");

      const err = new error.class(previousError);
      expect(err.message).toEqual(error.message);
      expect(err.previousError).toMatchObject(previousError);
    });
  });
});
