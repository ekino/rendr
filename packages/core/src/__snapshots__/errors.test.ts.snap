// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test errors Internal Server Error > should return custom message 1`] = `[InternalServerError: custom message]`;

exports[`test errors Normalization Error > should return custom message 1`] = `[NormalizationError: custom message]`;

exports[`test errors Not Found Error > should return custom message 1`] = `[NotFoundError: custom message]`;

exports[`test errors Not Found Exception > should return custom message 1`] = `[NotFoundError: custom message]`;
